vscode_extension/src/providers/AacDefinitionEditorProvider.ts

```diff
@@ -0,0 +1,235 @@
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { disposeAll } from '../disposable';
+import { AacDefinitionsDocument, AacDefinitionEdit, AacEditorEventTypes } from '../editor/definitions/DefinitionsDocument';
+import { getNonce } from '../nonce';
+import { aacRestApi, DefinitionModel } from "../requests/aacRequests";
+import { IncomingMessage } from 'http';
+
+/**
+ * Provider for AaC visual definition editors.
+ *
+ * AaC visual definition editors allow users to visually interact with the AaC REST API and definitions. There is no real underlying file for these editors as multiple definitions can exist in a file.
+ */
+export class AacDefinitionEditorProvider implements vscode.CustomEditorProvider<AacDefinitionsDocument> {
+
+    public static register(context: vscode.ExtensionContext): vscode.Disposable {
+        return vscode.window.registerCustomEditorProvider(
+            AacDefinitionEditorProvider.viewType,
+            new AacDefinitionEditorProvider(context),
+            {
+                webviewOptions: {
+                    retainContextWhenHidden: true, //keeps the webview alive even when it is not visible. Does have a memory overhead.
+                },
+                supportsMultipleEditorsPerDocument: false,
+            });
+    }
+
+    private static readonly viewType = 'aac.visualEditor';
+
+    /**
+     * Tracks all known webviews
+     */
+    private readonly webviews = new WebviewCollection();
+
+    constructor(
+        private readonly _context: vscode.ExtensionContext
+    ) { }
+
+    //#region CustomEditorProvider
```

Comment:
Is this actual syntax for some tool, or just a custom comment for us to know where the `CustomEditorProvider` is? The reason I ask is because ensuring we keep `CustomEditorProvider` changes within the `//#region`/`//#endregion` comments could be a bit challenging.

Reply:
I'll remove these, thank for catching that.